src/sinks/datadog/logs.rs

```diff
@@ -2,59 +2,123 @@ use crate::{
     config::{log_schema, DataType, SinkConfig, SinkContext, SinkDescription},
     event::Event,
     sinks::{
-        util::{self, encoding::EncodingConfig, tcp::TcpSink, Encoding, StreamSinkOld, UriSerde},
+        util::{
+            encode_event,
+            encoding::{EncodingConfig, EncodingConfiguration},
+            http::{BatchedHttpSink, HttpClient, HttpSink},
+            BatchConfig, BatchSettings, BoxedRawValue, Compression, Encoding, JsonArrayBuffer,
+            TowerRequestConfig, VecBuffer,
+        },
         Healthcheck, VectorSink,
     },
-    tls::{MaybeTlsSettings, TlsConfig},
 };
 use bytes::Bytes;
-use futures01::{stream::iter_ok, Sink};
+use flate2::write::GzEncoder;
+use futures::FutureExt;
+use futures01::Sink;
+use http::{Request, StatusCode};
+use hyper::body::Body;
 use serde::{Deserialize, Serialize};
+use serde_json::json;
+use std::io::Write;
 
-#[derive(Deserialize, Serialize, Debug)]
+#[derive(Deserialize, Serialize, Debug, Clone)]
 #[serde(deny_unknown_fields)]
 pub struct DatadogLogsConfig {
-    endpoint: Option<UriSerde>,
+    endpoint: Option<String>,
     api_key: String,
     encoding: EncodingConfig<Encoding>,
-    tls: Option<TlsConfig>,
+
+    #[serde(default)]
+    compression: Compression,
+
+    compression_level: Option<u32>,
+
+    #[serde(default)]
+    batch: BatchConfig,
+
+    #[serde(default)]
+    request: TowerRequestConfig,
+}
+
+#[derive(Clone)]
+pub struct DatadogLogsJsonService {
+    config: DatadogLogsConfig,
+}
+
+#[derive(Clone)]
+pub struct DatadogLogsTextService {
+    config: DatadogLogsConfig,
 }
 
 inventory::submit! {
     SinkDescription::new_without_default::<DatadogLogsConfig>("datadog_logs")
 }
 
+impl DatadogLogsConfig {
+    fn get_endpoint(&self) -> &str {
+        match &self.endpoint {
+            Some(ref endpoint) => &endpoint,
+            None => "https://http-intake.logs.datadoghq.eu/v1/input",
+        }
+    }
+}
+
 #[typetag::serde(name = "datadog_logs")]
 impl SinkConfig for DatadogLogsConfig {
     fn build(&self, cx: SinkContext) -> crate::Result<(VectorSink, Healthcheck)> {
-        let (host, port) = if let Some(uri) = &self.endpoint {
-            let host = uri
-                .host()
-                .ok_or_else(|| "A host is required for endpoint".to_string())?;
-            let port = uri
-                .port_u16()
-                .ok_or_else(|| "A port is required for endpoint".to_string())?;
-
-            (host.to_string(), port)
-        } else {
-            ("intake.logs.datadoghq.com".to_string(), 10516)
-        };
+        let request_settings = self.request.unwrap_with(&TowerRequestConfig::default());
+        let client = HttpClient::new(cx.resolver(), None)?;
 
-        let tls_settings = MaybeTlsSettings::from_config(
-            &Some(self.tls.clone().unwrap_or_else(TlsConfig::enabled)),
-            false,
-        )?;
```

Comment:
Note that the TLS settings are *not* obsolete here, as they can be used to control options other than turning on TLS mode — for example, certificates and keys.